test/ShapeCrawler.Tests.Unit/ShapeTests.cs

```diff
@@ -765,4 +767,129 @@ public void Name_Setter_sets_grouped_shape_name()
+    [TestCase("DownArrow", "[5.556,37.152]")]
+    [TestCase("StripedRightArrow", "[25.926,161.574]")]
+    [TestCase("NotchedRightArrow", "[55.556,143.056]")]
+    [TestCase("BentUpArrow", "[100,80.902,29.862]")]
+    [TestCase("LeftRightArrow", "[0,20.486]")]
+    [TestCase("UpDownArrow", "[41.666,27.778]")]
+    [TestCase("LeftUpArrow", "[10.562,70.422,59.156]")]
+    [TestCase("LeftRightUpArrow", "[0,20.422,50]")]
+    [TestCase("QuadArrow", "[0,75.986,8.38]")]
+    [TestCase("LeftArrowCallout", "[0,48.592,57.042,14.788]")]
+    [TestCase("RightArrowCallout", "[50,26.056,79.578,76.434]")]
+    [TestCase("UpArrowCallout", "[0,100,165.494,15.87]")]
+    [TestCase("DownArrowCallout", "[0,100,145.774,6.01]")]
+    [TestCase("UpDownArrowCallout", "[81.69,100,78.168,0]")]
+    [TestCase("QuadArrowCallout", "[0,0,62.97,0]")]
+    [TestCase("BentArrow", "[61.972,30.986,100,0]")]
+    [TestCase("UTurnArrow", "[0,18.31,85.212,5.81,135.212]")]
+    [TestCase("CircularArrow", "[77.376,6331.394,36798.45,9442.808,38.688]")]
+    //[TestCase("LeftCircularArrow", "[]")] // Malformed geometry. Formula has no value.
+    [TestCase("LeftRightCircularArrow", "[0,6773.122,1642.124,36866.022,8.998]")]
+    [TestCase("CurvedRightArrow", "[33.77,17.324,111.458]")]
+    [TestCase("CurvedLeftArrow", "[78.226,40.222,137.152]")]
+    [TestCase("CurvedUpArrow", "[1.734,42.154,33.68]")]
+    [TestCase("CurvedDownArrow", "[28.276,68.26,57.522]")]
+    [TestCase("SwooshArrow", "[147.778, 140]")]
+    [TestCase("CircularArrow", "[77.376,6331.394,36798.45,9442.808,38.688]")]
+    //[TestCase("LeftCircularArrow", "[]")] // Malformed geometry. Formula has no value.
+    [TestCase("LeftRightCircularArrow", "[0,6773.122,1642.124,36866.022,8.998]")]
+    [TestCase("CurvedRightArrow", "[33.77,17.324,111.458]")]
+    [TestCase("CurvedLeftArrow", "[78.226,40.222,137.152]")]
+    [TestCase("CurvedUpArrow", "[1.734,42.154,33.68]")]
+    [TestCase("CurvedDownArrow", "[28.276,68.26,57.522]")]
+    [TestCase("Cube", "[121.528]")]
+    [TestCase("Can", "[100]")]
+    [TestCase("Sun", "[93.75]")]
+    [TestCase("Moon", "[175]")]
+    // [TestCase("SmileyFace", "[]")] // Malformed geometry. Formula has no value.
+    [TestCase("FoldedCorner", "[100]")]
+    [TestCase("Bevel", "[11.266]")]
+    [TestCase("Frame", "[100]")]
+    [TestCase("HalfFrame", "[107.638, 17.36]")]
+    [TestCase("Corner", "[147.222,154.862]")]
+    [TestCase("DiagonalStripe", "[32.87]")]
+    [TestCase("Chord", "[12235.572,37755.642]")]
+    [TestCase("Arc", "[14144.04,9163.314]")]
+    [TestCase("LeftBracket", "[100]")]
+    [TestCase("RightBracket", "[0]")]
+
+    public void Adjustments_getter_returns_values(string name, string expectedAdjustmentsJson)
+    {
+        // Arrange
+        var shape = 
+            new Presentation(TestAsset("062_shape-adjustments.pptx"))
+                .Slides[0]
+                .Shapes
+                .GetByName(name)
+                as RootShape;
```

Comment:
`RootShape` is an `internal` class. Internals are exposed to the test project for exclusive situations, especially for use in Assert block.

Is there any way to avoid using `RootShape` in test?

Reply:
Yah. The solution will be to make it part of the public interface, right? I had originally hesitated on that, hoping to have a better public interface. But by adding this to `IGeometry`, then any `IShape` will do in this context.

Reply:
> The solution will be to make it part of the public interface, right?

Right.

Reply:
Ok, that's done in f9913d377ac3ce265314bd87575f65b81b008e09